src/Compatibility/Microsoft.DotNet.ApiCompatibility/ValidationSuppressions/SuppressionEngine.cs

```diff
@@ -0,0 +1,208 @@
+        /// <param name="right">Optional. The right operand in a APICompat error.</param>
+        public void AddSuppression(string? diagnosticId, string? target, string? left = null, string? right = null)
+        {
+            var suppressionToAdd = new Suppression()
+            {
+                DiagnosticId = diagnosticId,
+                Target = target,
+                Left = left,
+                Right = right
+            };
+            AddSuppression(suppressionToAdd);
+        }
+
+        /// <summary>
+        /// Adds a suppression to the collection.
+        /// </summary>
+        /// <param name="suppressionToAdd">The <see cref="Suppression"/> to be added.</param>
+        public void AddSuppression(Suppression suppressionToAdd)
+        {
+            _readerWriterLock.EnterUpgradeableReadLock();
+            try
+            {
+                if (!_validationSuppressions.Contains(suppressionToAdd))
+                {
+                    _readerWriterLock.EnterWriteLock();
+                    try
+                    {
+                        _validationSuppressions.Add(suppressionToAdd);
+                    }
+                    finally
+                    {
+                        _readerWriterLock.ExitWriteLock();
+                    }
+                }
+            }
+            finally
+            {
+                _readerWriterLock.ExitUpgradeableReadLock();
+            }
+        }
+
+        /// <summary>
+        /// Writes all suppressions in collection down to a file.
+        /// </summary>
+        /// <param name="validationSuppressionFile">The path to the file to be written.</param>
+        public void WriteSuppressionsToFile(string validationSuppressionFile)
+        {
+            XmlSerializer serializer = new XmlSerializer(typeof(Suppression[]));
+
+            using (Stream writer = GetWritableStream(validationSuppressionFile))
+            {
+                _readerWriterLock.EnterReadLock();
+                try
+                {
+                    serializer.Serialize(writer, _validationSuppressions.ToArray());
+                    AfterWrittingSuppressionsCallback(writer);
+                }
+                finally
+                {
+                    _readerWriterLock.ExitReadLock();
+                }
+            }
+        }
+
+        protected virtual void AfterWrittingSuppressionsCallback(Stream stream)
+        {
+            // Do nothing. Used for tests.
+        }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="SuppressionEngine"/> based on the contents of a given suppression file.
+        /// </summary>
+        /// <param name="validationSuppressionFile">The path to the suppressions file to be used for initialization.</param>
+        /// <returns>An instance of <see cref="SuppressionEngine"/>.</returns>
+        public static SuppressionEngine CreateFromSuppressionFile(string validationSuppressionFile)
```

Comment:
I would name this, `FromFile` that is the pattern we usually use at other places.

Reply:
I tried looking for precedent in the framework and find both instances present pretty much equally across the framework (half use FromFile, other half use CreateFromFile). No strong feeling from my side except that I lake all factory methods start with 'Create' which makes them easy to find when using intellisense, but if removing it and having only the `FromFile` as the signature is preferred I can change it.

Reply:
Ok, makes sense to have `Create` prefix but, why not we remove the `Supression` from the naming so that it is not redundant and that long?

Reply:
Sounds good to me, We can leave it as `CreateFromFile`